File: frontend/src/services/authService.js
import api from "../api/axios";


const register = async (userData) => {

    const response = await api.post(
        "/api/v1/auth/register",
        userData
    );

    return response.data;

};


const login = async (credentials) => {

    const response = await api.post(
        "/api/v1/auth/login",
        credentials
    );

    return response.data;

};


const refreshToken = async (refreshToken) => {

    const response = await api.post(
        "/api/v1/auth/refresh",
        {
            refreshToken
        }
    );

    return response.data;

};


const logout = async (refreshToken) => {

    const response = await api.post(
        "/api/v1/auth/logout",
        {
            refreshToken
        }
    );

    return response.data;

};


const getCurrentUser = async () => {

    const response = await api.get(
        "/api/v1/auth/me"
    );

    return response.data;

};


export default {
    register,
    login,
    refreshToken,
    logout,
    getCurrentUser
};

File: auth-service/src/main/java/com/logistics/auth/service/AuthService.java
package com.logistics.auth.service;

import com.logistics.auth.dto.request.LoginRequest;
import com.logistics.auth.dto.request.RegisterRequest;
import com.logistics.auth.dto.response.AuthResponse;
import com.logistics.auth.dto.request.RefreshTokenRequest;

public interface AuthService {

	void register(RegisterRequest request);

	AuthResponse login(LoginRequest request);

	AuthResponse refreshToken(RefreshTokenRequest request);

	void logout(RefreshTokenRequest request);


}